pkg/kubelet/kubelet.go

```diff
@@ -869,6 +870,17 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *api.Pod, container *api.Contain
 		return result, err
 	}
 
+	// Determine the final values of variables in steps:
+	//
+	// 1.  Determine the runtime value of all variables by resolving
+	//     alternate env var sources.
+	// 2.  Determine the final value of each variable by expanding `$var` and `${var}`
+	//     references to other variables; the sources of variables are the declared
+	//     variables of the container and the service environment variables.
+	// 3.  Create the container's environment in the order variables are declared
+
+	// Step 1: resolve alternate env var sources
+	tmpEnv := make(map[string]string)
 	for _, value := range container.Env {
```

Comment:
I think this method has a bug -- currently, you could have incorrect expansion of a var declared later in a container's environment, which is not supposed to be legal according to the spec.